From f4ef1914628de1c9853834cffdaee61ad3cff4cb Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Sat, 19 Jun 2010 21:13:10 +0100
Subject: [PATCH] You can't mix GTK2 and GTK3, so depend on gtk-3.0 versions of libraries

---
 configure.ac |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3d5657a..8cc80d4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,7 +52,7 @@ AC_CHECK_FUNC(XScreenSaverQueryInfo, [
 AC_SUBST(TYPING_BREAK)
 AC_SUBST(SCREENSAVER_LIBS)
 
-PKG_CHECK_MODULES(LIBCANBERRA_GTK, libcanberra-gtk, [
+PKG_CHECK_MODULES(LIBCANBERRA_GTK, libcanberra-gtk3, [
   CANBERRA_GTK=1
   AC_SUBST(LIBCANBERRA_GTK_CFLAGS)
   AC_SUBST(LIBCANBERRA_GTK_LIBS)
@@ -112,7 +112,7 @@ COMMON_MODULES="gtk+-3.0 >= 2.90.0 dnl
  gio-2.0 dnl
  gconf-2.0 dnl
  libxml-2.0 dnl
- gnome-desktop-2.0 >= 2.29.4"
+ gnome-desktop-3.0 >= 2.29.4"
 PKG_CHECK_MODULES(CAPPLET, $COMMON_MODULES)
 PKG_CHECK_MODULES(GNOMECC, $COMMON_MODULES libgnome-menu >= 2.10.1)
 PKG_CHECK_MODULES(GNOMECC_SHELL, $COMMON_MODULES libgnome-menu unique-1.0 gio-unix-2.0)
-- 
1.7.0.1

