diff -up gnome-control-center-2.21.2/capplets/appearance/appearance-desktop.c.background-location gnome-control-center-2.21.2/capplets/appearance/appearance-desktop.c
--- gnome-control-center-2.21.2/capplets/appearance/appearance-desktop.c.background-location	2007-11-17 23:39:06.000000000 -0500
+++ gnome-control-center-2.21.2/capplets/appearance/appearance-desktop.c	2007-11-17 23:39:39.000000000 -0500
@@ -1114,6 +1114,8 @@ desktop_init (AppearanceData *data,
                      NULL));
 
   gtk_file_chooser_set_select_multiple (data->wp_filesel, TRUE);
+  gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (data->wp_filesel),
+                                       "/usr/share/backgrounds");
   gtk_file_chooser_set_use_preview_label (data->wp_filesel, FALSE);
 
   pictures = g_get_user_special_dir (G_USER_DIRECTORY_PICTURES);
