--- control-center-2.18.0/capplets/sound/sound-properties-capplet.c.ellipsize-sound	2007-03-27 01:03:48.000000000 -0400
+++ control-center-2.18.0/capplets/sound/sound-properties-capplet.c	2007-03-27 01:02:45.000000000 -0400
@@ -869,6 +869,7 @@
 	g_object_unref (G_OBJECT (model)); 
 
 	renderer = gtk_cell_renderer_text_new ();
+        g_object_set (renderer, "ellipsize", PANGO_ELLIPSIZE_END, NULL);
 	gtk_cell_layout_pack_start (GTK_CELL_LAYOUT (device_widget), renderer, TRUE);
 	gtk_cell_layout_add_attribute (GTK_CELL_LAYOUT (device_widget), renderer,
 			"text", MIXER_DEVICE_MODEL_NAME_COLUMN);
