--- control-center-2.9.4/capplets/background/gnome-wp-capplet.c.filesel	2005-02-02 12:48:58.000000000 -0500
+++ control-center-2.9.4/capplets/background/gnome-wp-capplet.c	2005-02-02 12:49:27.000000000 -0500
@@ -1270,7 +1270,8 @@
 						  NULL);
   gtk_file_chooser_set_select_multiple (GTK_FILE_CHOOSER (capplet->filesel),
 					TRUE);
-
+  gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (capplet->filesel),
+                                      "/usr/share/backgrounds");
   gtk_file_chooser_set_use_preview_label (GTK_FILE_CHOOSER (capplet->filesel),
 					  FALSE);
 
