diff -u gnome-control-center-2.19.5/capplets/default-applications/gnome-da-capplet.c hacked/capplets/default-applications/gnome-da-capplet.c
--- gnome-control-center-2.19.5/capplets/default-applications/gnome-da-capplet.c	2007-07-09 09:36:29.000000000 -0400
+++ hacked/capplets/default-applications/gnome-da-capplet.c	2007-07-12 10:41:31.000000000 -0400
@@ -241,7 +241,7 @@
     gtk_widget_set_sensitive (capplet->new_win_radiobutton, has_net_remote);
     gtk_widget_set_sensitive (capplet->new_tab_radiobutton, has_net_remote);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->web_browser_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->web_browser_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->web_browser_command_label, is_custom_active);
     gtk_widget_set_sensitive (capplet->web_browser_terminal_checkbutton, is_custom_active);
 }
@@ -293,7 +293,7 @@
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (capplet->mail_reader_terminal_checkbutton),
 				  gconf_client_get_bool (capplet->gconf, DEFAULT_APPS_KEY_MAILER_NEEDS_TERM, NULL));
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->mail_reader_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->mail_reader_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->mail_reader_command_label, is_custom_active);
     gtk_widget_set_sensitive (capplet->mail_reader_terminal_checkbutton, is_custom_active);
 }
@@ -344,7 +344,7 @@
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (capplet->media_player_terminal_checkbutton),
 				  gconf_client_get_bool (capplet->gconf, DEFAULT_APPS_KEY_MEDIA_NEEDS_TERM, NULL));
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->media_player_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->media_player_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->media_player_command_label, is_custom_active);
     gtk_widget_set_sensitive (capplet->media_player_terminal_checkbutton, is_custom_active);
 }
@@ -396,9 +396,9 @@
     g_free (terminal_cmd);
     g_free (terminal_cmd_arg);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->terminal_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->terminal_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->terminal_command_label, is_custom_active);
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->terminal_exec_flag_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->terminal_exec_flag_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->terminal_exec_flag_label, is_custom_active);
 }
 
@@ -447,7 +447,7 @@
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (capplet->visual_startup_checkbutton),
 				  gconf_client_get_bool (capplet->gconf, DEFAULT_APPS_KEY_VISUAL_STARTUP, NULL));
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->visual_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->visual_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->visual_command_label, is_custom_active);
 
 }
@@ -497,7 +497,7 @@
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (capplet->mobility_startup_checkbutton),
 				  gconf_client_get_bool (capplet->gconf, DEFAULT_APPS_KEY_MOBILITY_STARTUP, NULL));
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->mobility_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->mobility_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->mobility_command_label, is_custom_active);
 }
 
@@ -628,7 +628,7 @@
     /* TODO: Remove when GConfPropertyEditor will be used */
     gtk_entry_set_text (GTK_ENTRY (capplet->web_browser_command_entry), command);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->web_browser_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->web_browser_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->web_browser_command_label, is_custom_active);
     gtk_widget_set_sensitive (capplet->web_browser_terminal_checkbutton, is_custom_active);
 
@@ -704,7 +704,7 @@
 
     gtk_entry_set_text (GTK_ENTRY (capplet->mail_reader_command_entry), command);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->mail_reader_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->mail_reader_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->mail_reader_command_label, is_custom_active);
     gtk_widget_set_sensitive (capplet->mail_reader_terminal_checkbutton, is_custom_active);
 
@@ -733,7 +733,7 @@
 
     gtk_entry_set_text (GTK_ENTRY (capplet->media_player_command_entry), command);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->media_player_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->media_player_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->media_player_command_label, is_custom_active);
     gtk_widget_set_sensitive (capplet->media_player_terminal_checkbutton, is_custom_active);
 
@@ -762,9 +762,9 @@
 
     gtk_entry_set_text (GTK_ENTRY (capplet->terminal_command_entry), command);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->terminal_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->terminal_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->terminal_command_label, is_custom_active);
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->terminal_exec_flag_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->terminal_exec_flag_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->terminal_exec_flag_label, is_custom_active);
 
     if (gtk_combo_box_get_active (GTK_COMBO_BOX (capplet->term_combo_box)) != index)
@@ -792,7 +792,7 @@
 
     gtk_entry_set_text (GTK_ENTRY (capplet->visual_command_entry), command);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->visual_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->visual_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->visual_command_label, is_custom_active);
 
     if (gtk_combo_box_get_active (GTK_COMBO_BOX (capplet->visual_combo_box)) != index)
@@ -821,7 +821,7 @@
 
     gtk_entry_set_text (GTK_ENTRY (capplet->mobility_command_entry), command);
 
-    gtk_editable_set_editable (GTK_EDITABLE (capplet->mobility_command_entry), is_custom_active);
+    gtk_widget_set_sensitive (capplet->mobility_command_entry, is_custom_active);
     gtk_widget_set_sensitive (capplet->mobility_command_label, is_custom_active);
 
     if (gtk_combo_box_get_active (GTK_COMBO_BOX (capplet->mobility_combo_box)) != index)
diff -u gnome-control-center-2.19.5/capplets/default-applications/gnome-default-applications-properties.glade hacked/capplets/default-applications/gnome-default-applications-properties.glade
--- gnome-control-center-2.19.5/capplets/default-applications/gnome-default-applications-properties.glade	2007-04-23 10:58:28.000000000 -0400
+++ hacked/capplets/default-applications/gnome-default-applications-properties.glade	2007-07-12 10:38:38.000000000 -0400
@@ -167,7 +167,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -450,7 +450,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -633,7 +633,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -852,7 +852,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -1034,7 +1034,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -1217,7 +1217,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -1434,7 +1434,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -1616,7 +1616,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -1666,7 +1666,7 @@
 				<widget class="GtkEntry" id="terminal_exec_flag_entry">
 				  <property name="visible">True</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -1880,7 +1880,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
@@ -2059,7 +2059,7 @@
 				  <property name="visible">True</property>
 				  <property name="tooltip" translatable="yes">All %s occurrences will be replaced with actual link</property>
 				  <property name="can_focus">True</property>
-				  <property name="editable">False</property>
+				  <property name="editable">True</property>
 				  <property name="visibility">True</property>
 				  <property name="max_length">0</property>
 				  <property name="text" translatable="yes"></property>
