diff -up gnome-control-center-2.21.4/capplets/default-applications/gnome-default-applications.xml.in.default-apps gnome-control-center-2.21.4/capplets/default-applications/gnome-default-applications.xml.in
--- gnome-control-center-2.21.4/capplets/default-applications/gnome-default-applications.xml.in.default-apps	2007-12-20 17:00:56.000000000 -0500
+++ gnome-control-center-2.21.4/capplets/default-applications/gnome-default-applications.xml.in	2007-12-21 12:23:29.000000000 -0500
@@ -44,52 +44,12 @@
       <win-command>galeon -w %s</win-command>
     </web-browser>
     <web-browser>
-      <_name>Encompass</_name>
-      <executable>encompass</executable>
-      <command>encompass %s</command>
-      <icon-name>encompass</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>false</netscape-remote>
-    </web-browser>
-    <web-browser>
-      <_name>Firebird</_name>
-      <executable>mozilla-firebird</executable>
-      <command>mozilla-firebird %s</command>
-      <icon-name></icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>mozilla-firebird -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>mozilla-firebird -remote "openurl(%s,new-window)"</win-command>
-    </web-browser>
-    <web-browser>
       <_name>Firefox</_name>
       <executable>firefox</executable>
       <command>firefox %s</command>
       <icon-name>firefox</icon-name>
       <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>firefox -new-tab "%s"</tab-command>
-      <win-command>firefox -new-window "%s"</win-command>
-    </web-browser>
-    <web-browser>
-      <_name>Iceweasel</_name>
-      <executable>iceweasel</executable>
-      <command>iceweasel %s</command>
-      <icon-name>iceweasel</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>iceweasel -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>iceweasel -remote "openurl(%s,new-window)"</win-command>
-    </web-browser>
-    <web-browser>
-      <_name>Mozilla 1.6</_name>
-      <executable>mozilla-1.6</executable>
-      <command>mozilla-1.6 %s</command>
-      <icon-name>mozilla-icon</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>mozilla-1.6 -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>mozilla-1.6 -remote "openurl(%s,new-window)"</win-command>
+      <netscape-remote>false</netscape-remote>
     </web-browser>
     <web-browser>
       <_name>Mozilla</_name>
@@ -97,9 +57,7 @@
       <command>mozilla %s</command>
       <icon-name>mozilla-icon</icon-name>
       <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>mozilla -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>mozilla -remote "openurl(%s,new-window)"</win-command>
+      <netscape-remote>false</netscape-remote>
     </web-browser>
     <web-browser>
       <_name>SeaMonkey</_name>
@@ -112,24 +70,12 @@
       <win-command>seamonkey -remote "openurl(%s,new-window)"</win-command>
     </web-browser>
     <web-browser>
-      <_name>Iceape</_name>
-      <executable>iceape</executable>
-      <command>iceape %s</command>
-      <icon-name>iceape</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>iceape -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>iceape -remote "openurl(%s,new-window)"</win-command>
-    </web-browser>
-    <web-browser>
-      <_name>Netscape Communicator</_name>
+      <_name>Netscape</_name>
       <executable>netscape</executable>
       <command>netscape %s</command>
       <icon-name>netscape</icon-name>
       <run-in-terminal>false</run-in-terminal>
-      <netscape-remote>true</netscape-remote>
-      <tab-command>netscape -remote "openurl(%s,new-tab)"</tab-command>
-      <win-command>netscape -remote "openurl(%s,new-window)"</win-command>
+      <netscape-remote>false</netscape-remote>
     </web-browser>
     <web-browser>
       <_name>Konqueror</_name>
@@ -153,7 +99,7 @@
     <mail-reader>
       <_name>Evolution Mail Reader</_name>
       <executable>evolution</executable>
-      <command>evolution %s</command>
+      <command>evolution --component=mail %s</command>
       <icon-name>evolution</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
@@ -179,27 +125,13 @@
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
-      <_name>Thunderbird</_name>
+      <_name>Thunderbird Mail</_name>
       <executable>thunderbird</executable>
       <command>thunderbird %s</command>
       <icon-name>thunderbird</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
     <mail-reader>
-      <_name>Mozilla Thunderbird</_name>
-      <executable>mozilla-thunderbird</executable>
-      <command>mozilla-thunderbird %s</command>
-      <icon-name>thunderbird</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-    </mail-reader>
-    <mail-reader>
-      <_name>Mozilla Mail</_name>
-      <executable>mozilla</executable>
-      <command>mozilla -mail %s</command>
-      <icon-name>mozilla-mail-icon</icon-name>
-      <run-in-terminal>false</run-in-terminal>
-    </mail-reader>
-    <mail-reader>
       <_name>SeaMonkey Mail</_name>
       <executable>seamonkey</executable>
       <command>seamonkey -mail %s</command>
@@ -252,6 +191,20 @@
       <exec-flag>-e</exec-flag>
     </terminal>
     <terminal>
+      <_name>KDE Konsole</_name>
+      <executable>konsole</executable>
+      <command>konsole</command>
+      <icon-name>konsole</icon-name>
+      <exec-flag>-e</exec-flag>
+    </terminal>
+    <terminal>
+      <_name>KTerm</_name>
+      <executable>kterm</executable>
+      <command>kterm</command>
+      <icon-name>kterm</icon-name>
+      <exec-flag>-e</exec-flag>
+    </terminal>
+    <terminal>
       <_name>GNOME Terminal</_name>
       <executable>gnome-terminal</executable>
       <command>gnome-terminal</command>
