diff -up gnome-control-center-2.25.1/capplets/display/display-capplet.glade.display-no-help gnome-control-center-2.25.1/capplets/display/display-capplet.glade
--- gnome-control-center-2.25.1/capplets/display/display-capplet.glade.display-no-help	2008-11-05 09:21:32.000000000 -0500
+++ gnome-control-center-2.25.1/capplets/display/display-capplet.glade	2008-11-13 14:28:06.000000000 -0500
@@ -33,7 +33,7 @@
 
 	  <child>
 	    <widget class="GtkButton" id="helpbutton1">
-	      <property name="visible">True</property>
+	      <property name="visible">False</property>
 	      <property name="can_default">True</property>
 	      <property name="can_focus">True</property>
 	      <property name="label">gtk-help</property>
