diff -up gnome-control-center-2.26.0/capplets/display/display-capplet.glade.mnemonics gnome-control-center-2.26.0/capplets/display/display-capplet.glade
--- gnome-control-center-2.26.0/capplets/display/display-capplet.glade.mnemonics	2009-04-09 22:17:08.628091488 -0400
+++ gnome-control-center-2.26.0/capplets/display/display-capplet.glade	2009-04-09 22:17:16.344090972 -0400
@@ -262,6 +262,7 @@
 		  <property name="width_chars">-1</property>
 		  <property name="single_line_mode">False</property>
 		  <property name="angle">0</property>
+		  <property name="mnemonic-widget">resolution_combo</property>
 		</widget>
 		<packing>
 		  <property name="left_attach">1</property>
@@ -290,6 +291,7 @@
 		  <property name="width_chars">-1</property>
 		  <property name="single_line_mode">False</property>
 		  <property name="angle">0</property>
+		  <property name="mnemonic-widget">refresh_combo</property>
 		</widget>
 		<packing>
 		  <property name="left_attach">1</property>
@@ -318,6 +320,7 @@
 		  <property name="width_chars">-1</property>
 		  <property name="single_line_mode">False</property>
 		  <property name="angle">0</property>
+		  <property name="mnemonic-widget">rotation_combo</property>
 		</widget>
 		<packing>
 		  <property name="left_attach">3</property>
