diff -up gnome-control-center-2.28.1/capplets/about-me/gnome-about-me-fingerprint.c.finger-icons gnome-control-center-2.28.1/capplets/about-me/gnome-about-me-fingerprint.c
--- gnome-control-center-2.28.1/capplets/about-me/gnome-about-me-fingerprint.c.finger-icons	2009-10-26 13:02:36.578802754 -0400
+++ gnome-control-center-2.28.1/capplets/about-me/gnome-about-me-fingerprint.c	2009-10-26 13:03:11.893802706 -0400
@@ -468,7 +468,7 @@ assistant_prepare (GtkAssistant *ass, Gt
 		{
 			char *filename;
 
-			filename = g_strdup_printf ("%s.png", data->finger);
+			filename = g_strdup_printf ("%s.svg", data->finger);
 			path = g_build_filename (GNOMECC_PIXMAP_DIR, filename, NULL);
 			g_free (filename);
 		}
