diff -urNp gnome-control-center-2.91.0.old/panels/datetime/Makefile.am gnome-control-center-2.91.0/panels/datetime/Makefile.am
--- gnome-control-center-2.91.0.old/panels/datetime/Makefile.am	2010-10-06 17:23:04.016807000 +0100
+++ gnome-control-center-2.91.0/panels/datetime/Makefile.am	2010-10-06 17:30:21.541806993 +0100
@@ -68,17 +68,17 @@ INCLUDES = 						\
 	-DDATADIR="\"$(uidir)\""			\
 	$(NULL)
 
-noinst_PROGRAMS = test-timezone
+#noinst_PROGRAMS = test-timezone
 
-test_timezone_SOURCES = test-timezone.c tz.c tz.h
-test_timezone_LDADD = $(PANEL_LIBS)
-test_timezone_CFLAGS = $(PANEL_CFLAGS)
+#test_timezone_SOURCES = test-timezone.c tz.c tz.h
+#test_timezone_LDADD = $(PANEL_LIBS)
+#test_timezone_CFLAGS = $(PANEL_CFLAGS)
 
 all: check
 
 # FIXME remove "|| :" when we have all the necessary pixmaps
-check: test-timezone
-	$(builddir)/test-timezone $(srcdir)/data || :
+#check: test-timezone
+#	$(builddir)/test-timezone $(srcdir)/data || :
 
 ccpanelsdir = $(PANELS_DIR)
 ccpanels_LTLIBRARIES = libdate_time.la
