--- gnome-control-center-2.28.0/capplets/appearance/appearance-desktop.c	2009-09-28 20:01:16.896708487 -0400
+++ hacked/capplets/appearance/appearance-desktop.c	2009-09-28 19:56:21.977442977 -0400
@@ -1246,8 +1246,6 @@
   GtkWidget *widget, *box, *button;
   char *url;
 
-  g_object_set (gtk_settings_get_default (), "gtk-tooltip-timeout", 500, NULL);
-
   data->wp_update_gconf = TRUE;
 
   data->wp_uris = NULL;
@@ -1387,6 +1385,7 @@
                     (GCallback) wp_props_wp_selected, data);
   g_signal_connect (data->wp_view, "query-tooltip",
                     (GCallback) wp_view_tooltip_cb, data);
+  gtk_widget_set_has_tooltip (data->wp_view, TRUE);
 
   wp_set_sensitivities (data);
 
