Index: capplets/sound/sound-theme.c
===================================================================
--- capplets/sound/sound-theme.c	(revision 9054)
+++ capplets/sound/sound-theme.c	(working copy)
@@ -509,6 +509,9 @@
 	if (data_dir != NULL)
 		gtk_file_chooser_add_shortcut_folder (GTK_FILE_CHOOSER (chooser), data_dir, NULL);
 
+#define SOUND_DATA_DIR "/usr/share/sounds/"
+	gtk_file_chooser_set_current_folder (GTK_FILE_CHOOSER (chooser), SOUND_DATA_DIR);
+
 	response = gtk_dialog_run (GTK_DIALOG (chooser));
 	filename = NULL;
 	if (response == GTK_RESPONSE_ACCEPT)
