diff --git a/capplets/about-me/gnome-about-me.c b/capplets/about-me/gnome-about-me.c
index 8119c3c..b7fe5d8 100644
--- a/capplets/about-me/gnome-about-me.c
+++ b/capplets/about-me/gnome-about-me.c
@@ -40,7 +40,6 @@
 #endif
 
 #include "e-image-chooser.h"
-#include "gnome-about-me-password.h"
 #include "gnome-about-me-fingerprint.h"
 #include "marshal.h"
 
@@ -861,10 +860,7 @@ about_me_button_clicked_cb (GtkDialog *dialog, gint response_id, GnomeAboutMe *m
 static void
 about_me_passwd_clicked_cb (GtkWidget *button, GnomeAboutMe *me)
 {
-	GtkBuilder *dialog;
-
-	dialog = me->dialog;
-	gnome_about_me_password (GTK_WINDOW (WID ("about-me-dialog")));
+	g_spawn_command_line_async ("/usr/bin/userpasswd", NULL);
 }
 
 static void
