diff -up gnome-control-center-2.91.4/panels/sound/gvc-balance-bar.c.update-policy gnome-control-center-2.91.4/panels/sound/gvc-balance-bar.c
--- gnome-control-center-2.91.4/panels/sound/gvc-balance-bar.c.update-policy	2011-01-09 17:49:51.242902003 -0500
+++ gnome-control-center-2.91.4/panels/sound/gvc-balance-bar.c	2011-01-09 17:50:07.041902021 -0500
@@ -141,7 +141,6 @@ _scale_box_new (GvcBalanceBar *bar)
         bar->priv->end_box = ebox = gtk_hbox_new (FALSE, 6);
         gtk_box_pack_start (GTK_BOX (box), ebox, FALSE, FALSE, 0);
 
-        gtk_range_set_update_policy (GTK_RANGE (priv->scale), GTK_UPDATE_CONTINUOUS);
         ca_gtk_widget_disable_sounds (bar->priv->scale, FALSE);
         gtk_widget_add_events (bar->priv->scale, GDK_SCROLL_MASK);
 
diff -up gnome-control-center-2.91.4/panels/sound/gvc-channel-bar.c.update-policy gnome-control-center-2.91.4/panels/sound/gvc-channel-bar.c
--- gnome-control-center-2.91.4/panels/sound/gvc-channel-bar.c.update-policy	2011-01-09 17:50:21.403901935 -0500
+++ gnome-control-center-2.91.4/panels/sound/gvc-channel-bar.c	2011-01-09 17:50:31.912902006 -0500
@@ -158,7 +158,6 @@ _scale_box_new (GvcChannelBar *bar)
                 gtk_box_pack_start (GTK_BOX (ebox), priv->mute_box, FALSE, FALSE, 0);
         }
 
-        gtk_range_set_update_policy (GTK_RANGE (priv->scale), GTK_UPDATE_CONTINUOUS);
         ca_gtk_widget_disable_sounds (bar->priv->scale, FALSE);
         gtk_widget_add_events (bar->priv->scale, GDK_SCROLL_MASK);
 
