Index: capplets/keybindings/gnome-keybinding-properties.c
===================================================================
--- capplets/keybindings/gnome-keybinding-properties.c	(revision 8611)
+++ capplets/keybindings/gnome-keybinding-properties.c	(working copy)
@@ -711,6 +711,7 @@
   keys = (KeyListEntry *) g_array_free (keylist->entries, FALSE);
   if (keylist->package)
     {
+      bind_textdomain_codeset (keylist->package, "UTF-8");
       title = dgettext (keylist->package, keylist->name);
     } else {
       title = _(keylist->name);
