diff -up gnome-control-center-2.27.3/capplets/keyboard/gnome-keyboard-properties-xkb.c.libgnomekbd gnome-control-center-2.27.3/capplets/keyboard/gnome-keyboard-properties-xkb.c
--- gnome-control-center-2.27.3/capplets/keyboard/gnome-keyboard-properties-xkb.c.libgnomekbd	2009-07-01 13:56:06.628794369 -0400
+++ gnome-control-center-2.27.3/capplets/keyboard/gnome-keyboard-properties-xkb.c	2009-07-01 13:56:19.537797461 -0400
@@ -161,8 +161,7 @@ setup_xkb_tabs (GladeXML * dialog, GConf
 	gkbd_desktop_config_init (&desktop_config, xkb_gconf_client, engine);
 	gkbd_desktop_config_load_from_gconf (&desktop_config);
 
-	xkl_config_registry_load (config_registry,
-				  desktop_config.load_extra_items);
+	xkl_config_registry_load (config_registry, FALSE);
 
 	gkbd_keyboard_config_init (&initial_config, xkb_gconf_client,
 				   engine);
