diff -up gnome-control-center-2.23.2/capplets/keybindings/eggaccelerators.c.marshalers gnome-control-center-2.23.2/capplets/keybindings/eggaccelerators.c
diff -up gnome-control-center-2.23.2/capplets/keybindings/eggcellrendererkeys.c.marshalers gnome-control-center-2.23.2/capplets/keybindings/eggcellrendererkeys.c
--- gnome-control-center-2.23.2/capplets/keybindings/eggcellrendererkeys.c.marshalers	2008-06-03 09:03:15.000000000 -0400
+++ gnome-control-center-2.23.2/capplets/keybindings/eggcellrendererkeys.c	2008-06-03 08:48:51.000000000 -0400
@@ -1,5 +1,6 @@
 #include <config.h>
 #include <libintl.h>
+#include <gtk/gtkmarshal.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkx.h>
 #include <gdk/gdkkeysyms.h>
