diff -pruN gnome-control-center-2.21.2.orig/gnome-settings-daemon/gnome-settings-xsettings.c gnome-control-center-2.21.2/gnome-settings-daemon/gnome-settings-xsettings.c
--- gnome-control-center-2.21.2.orig/gnome-settings-daemon/gnome-settings-xsettings.c	2007-07-17 18:42:56.000000000 +0900
+++ gnome-control-center-2.21.2/gnome-settings-daemon/gnome-settings-xsettings.c	2007-12-18 13:49:53.000000000 +0900
@@ -225,6 +225,8 @@ static TranslationEntry translations [] 
 	  GCONF_VALUE_STRING,	translate_string_string },
 	{ "/desktop/gnome/interface/gtk-im-status-style",	"Gtk/IMStatusStyle",
 	  GCONF_VALUE_STRING,	translate_string_string },
+	{ "/desktop/gnome/interface/gtk-im-module",		"Gtk/IMModule",
+	  GCONF_VALUE_STRING,	translate_string_string },
 	{ "/desktop/gnome/interface/icon_theme",		"Net/IconThemeName",
 	  GCONF_VALUE_STRING,	translate_string_string },
 	{ "/desktop/gnome/interface/file_chooser_backend",	"Gtk/FileChooserBackend",
