Index: capplets/mouse/gnome-mouse-properties.c
===================================================================
--- capplets/mouse/gnome-mouse-properties.c	(revision 8830)
+++ capplets/mouse/gnome-mouse-properties.c	(working copy)
@@ -495,7 +495,7 @@
 			g_free (page_name);
 		}
 
-		capplet_set_icon (dialog_win, "gnome-dev-mouse-optical");
+		capplet_set_icon (dialog_win, "input-mouse");
 		gtk_widget_show (dialog_win);
 
 		gtk_main ();
Index: capplets/mouse/gnome-settings-mouse.desktop.in.in
===================================================================
--- capplets/mouse/gnome-settings-mouse.desktop.in.in	(revision 8830)
+++ capplets/mouse/gnome-settings-mouse.desktop.in.in	(working copy)
@@ -2,7 +2,7 @@
 _Name=Mouse
 _Comment=Set your mouse preferences
 Exec=gnome-mouse-properties
-Icon=gnome-dev-mouse-optical
+Icon=input-mouse
 Terminal=false
 Type=Application
 StartupNotify=true
