--- gnome-control-center-2.28.1/capplets/common/wm-common.c	2009-09-21 06:44:55.000000000 -0400
+++ hacked/capplets/common/wm-common.c	2009-12-08 21:31:52.318541572 -0500
@@ -27,7 +27,7 @@
   guchar *val;
 
   if (wm_window == None)
-    return g_strdup (WM_COMMON_UNKNOWN);
+    return NULL;
 
   utf8_string = XInternAtom (GDK_DISPLAY (), "UTF8_STRING", False);
 
@@ -46,7 +46,7 @@
       type != utf8_string || format != 8 || nitems == 0 ||
       !g_utf8_validate (val, nitems, NULL))
     {
-      retval = g_strdup (WM_COMMON_UNKNOWN);
+      retval = NULL;
     }
   else
     {
@@ -93,7 +93,7 @@
       char *wm_name = wm_common_get_window_manager_property (wm_atom);
       char *to_copy[] = { NULL, NULL };
 
-      to_copy[0] = wm_name;
+      to_copy[0] = wm_name ? wm_name : WM_COMMON_UNKNOWN;
 
       results = g_strdupv (to_copy);
       g_free (wm_name);
