diff -up gnome-control-center-2.21.5/capplets/appearance/data/appearance.glade.no-fonts gnome-control-center-2.21.5/capplets/appearance/data/appearance.glade
--- gnome-control-center-2.21.5/capplets/appearance/data/appearance.glade.no-fonts	2008-01-22 11:30:59.000000000 -0500
+++ gnome-control-center-2.21.5/capplets/appearance/data/appearance.glade	2008-01-22 11:38:08.000000000 -0500
@@ -619,19 +619,6 @@
               </widget>
             </child>
             <child>
-              <widget class="GtkButton" id="go_to_font_button">
-                <property name="visible">True</property>
-                <property name="can_focus">True</property>
-                <property name="can_default">True</property>
-                <property name="label" translatable="yes">Go _to Fonts Folder</property>
-                <property name="use_underline">True</property>
-                <property name="response_id">1</property>
-              </widget>
-              <packing>
-                <property name="position">1</property>
-              </packing>
-            </child>
-            <child>
               <widget class="GtkButton" id="button3">
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
@@ -641,7 +628,7 @@
                 <property name="response_id">-7</property>
               </widget>
               <packing>
-                <property name="position">2</property>
+                <property name="position">1</property>
               </packing>
             </child>
           </widget>
diff -up gnome-control-center-2.21.5/capplets/appearance/appearance-font.c.no-fonts gnome-control-center-2.21.5/capplets/appearance/appearance-font.c
--- gnome-control-center-2.21.5/capplets/appearance/appearance-font.c.no-fonts	2008-01-22 11:31:23.000000000 -0500
+++ gnome-control-center-2.21.5/capplets/appearance/appearance-font.c	2008-01-22 11:37:48.000000000 -0500
@@ -805,9 +805,6 @@ cb_details_response (GtkDialog *dialog, 
     capplet_help (GTK_WINDOW (dialog),
 		  "user-guide.xml",
 		  "goscustdesk-38");
-  } else if (response_id == 1) {
-    /* "Go to font folder" was clicked */
-    g_spawn_command_line_async ("nautilus --no-desktop fonts:///", NULL);
   } else
     gtk_widget_hide (GTK_WIDGET (dialog));
 }
@@ -817,20 +814,11 @@ cb_show_details (GtkWidget *button,
 		 AppearanceData *data)
 {
   if (!data->font_details) {
-    GnomeVFSURI *uri;
     GtkAdjustment *adjustment;
     GtkWidget *widget;
     EnumGroup *group;
 
     data->font_details = glade_xml_get_widget (data->xml, "render_details");
-    uri = gnome_vfs_uri_new ("fonts:///");
-    widget = glade_xml_get_widget (data->xml, "go_to_font_button");
-    if (uri == NULL) {
-      gtk_widget_hide (widget);
-    } else {
-      gnome_vfs_uri_unref (uri);
-      gtk_widget_show (widget);
-    }
 
     gtk_window_set_transient_for (GTK_WINDOW (data->font_details),
 				  GTK_WINDOW (glade_xml_get_widget (data->xml, "appearance_window")));
