diff -up gnome-control-center-2.20.1/capplets/default-applications/gnome-default-applications.xml.in.orca-command gnome-control-center-2.20.1/capplets/default-applications/gnome-default-applications.xml.in
--- gnome-control-center-2.20.1/capplets/default-applications/gnome-default-applications.xml.in.orca-command	2007-10-24 19:17:57.000000000 -0400
+++ gnome-control-center-2.20.1/capplets/default-applications/gnome-default-applications.xml.in	2007-10-24 19:18:12.000000000 -0400
@@ -306,7 +306,7 @@
     <visual>
       <_name>Orca with Magnifier</_name>
       <executable>orca</executable>
-      <command>orca -m</command>
+      <command>orca -e magnifier</command>
       <icon-name>orca</icon-name>
       <run-at-start>false</run-at-start>
     </visual>
