diff -up gnome-control-center-2.27.4/capplets/appearance/data/appearance.ui.save_as gnome-control-center-2.27.4/capplets/appearance/data/appearance.ui
--- gnome-control-center-2.27.4/capplets/appearance/data/appearance.ui.save_as	2009-08-03 11:41:07.595008046 -0400
+++ gnome-control-center-2.27.4/capplets/appearance/data/appearance.ui	2009-08-03 11:41:27.273986675 -0400
@@ -740,6 +740,7 @@
                         <property name="visible">True</property>
                         <property name="sensitive">False</property>
                         <property name="can_focus">True</property>
+                        <property name="use-underline">True</property>
                         <property name="receives_default">False</property>
                         <property name="events">GDK_POINTER_MOTION_MASK | GDK_POINTER_MOTION_HINT_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK</property>
                       </object>
