diff -up gnome-control-center-2.27.91/shell/control-center.schemas.in.common-tasks gnome-control-center-2.27.91/shell/control-center.schemas.in
--- gnome-control-center-2.27.91/shell/control-center.schemas.in.common-tasks	2009-09-09 20:01:18.076653434 -0400
+++ gnome-control-center-2.27.91/shell/control-center.schemas.in	2009-09-09 20:02:12.237908583 -0400
@@ -7,7 +7,7 @@
 			<type>list</type>
 			<list_type>string</list_type>
 			<locale name="C">
-				<default><!-- Translators: The format of this string is the task name to be displayed (translate that part) followed by a ";" separator then the filename (DONT translate the file name) of a .desktop file to launch. Multiple entries are separated by a "," -->[Change Theme;gtk-theme-selector.desktop,Set Preferred Applications;default-applications.desktop,Add Printer;gnome-cups-manager.desktop]</default>
+				<default><!-- Translators: The format of this string is the task name to be displayed (translate that part) followed by a ";" separator then the filename (DONT translate the file name) of a .desktop file to launch. Multiple entries are separated by a "," -->[Change Theme;gnome-appearance-properties.desktop,Set Preferred Applications;gnome-default-applications.desktop,Add Printer;system-config-printer.desktop]</default>
 				<short>Task names and associated .desktop files</short>
 				<long>The task name to be displayed in the control-center followed by a ";" separator then the filename of an associated .desktop file to launch for that task.</long>
 			</locale>
