diff -up gnome-control-center-2.91.90/panels/sound/data/gnome-sound-applet.desktop.in.if-session gnome-control-center-2.91.90/panels/sound/data/gnome-sound-applet.desktop.in
--- gnome-control-center-2.91.90/panels/sound/data/gnome-sound-applet.desktop.in.if-session	2011-02-28 13:55:50.475339826 -0500
+++ gnome-control-center-2.91.90/panels/sound/data/gnome-sound-applet.desktop.in	2011-02-28 13:55:59.439227758 -0500
@@ -14,4 +14,4 @@ X-GNOME-Bugzilla-Component=sound
 # See http://bugzilla.gnome.org/show_bug.cgi?id=568320
 #X-GNOME-Autostart-Phase=Panel
 X-GNOME-Autostart-Notify=true
-AutostartCondition=GNOME3 is-session gnome-fallback
+AutostartCondition=GNOME3 if-session gnome-fallback
