diff -up gnome-control-center-2.23.4/capplets/accessibility/at-properties/at-properties.desktop.in.in.standard-icon gnome-control-center-2.23.4/capplets/accessibility/at-properties/at-properties.desktop.in.in
--- gnome-control-center-2.23.4/capplets/accessibility/at-properties/at-properties.desktop.in.in.standard-icon	2008-06-20 13:43:47.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/accessibility/at-properties/at-properties.desktop.in.in	2008-06-20 13:44:30.000000000 -0400
@@ -2,7 +2,7 @@
 _Name=Assistive Technologies
 _Comment=Enable support for GNOME assistive technologies at login
 Exec=gnome-at-properties
-Icon=gnome-settings-accessibility-technologies
+Icon=preferences-desktop-accessibility
 Terminal=false
 Type=Application
 StartupNotify=true
diff -up gnome-control-center-2.23.4/capplets/accessibility/at-properties/main.c.standard-icon gnome-control-center-2.23.4/capplets/accessibility/at-properties/main.c
--- gnome-control-center-2.23.4/capplets/accessibility/at-properties/main.c.standard-icon	2008-06-20 13:44:46.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/accessibility/at-properties/main.c	2008-06-20 13:45:08.000000000 -0400
@@ -188,7 +188,7 @@ setup_dialog (GladeXML *dialog)
 			  G_CALLBACK (cb_login_preferences), NULL);
 
 	widget = WID ("at_properties_dialog");
-	capplet_set_icon (widget, "gnome-settings-accessibility-technologies");
+	capplet_set_icon (widget, "preferences-desktop-accessibility");
 
 	g_signal_connect (G_OBJECT (widget),
 			  "response",
diff -up gnome-control-center-2.23.4/capplets/appearance/appearance-main.c.standard-icon gnome-control-center-2.23.4/capplets/appearance/appearance-main.c
--- gnome-control-center-2.23.4/capplets/appearance/appearance-main.c.standard-icon	2008-06-20 14:04:35.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/appearance/appearance-main.c	2008-06-20 14:04:55.000000000 -0400
@@ -156,7 +156,7 @@ main (int argc, char **argv)
 
   /* prepare the main window */
   w = glade_xml_get_widget (data->xml, "appearance_window");
-  capplet_set_icon (w, "gnome-settings-theme");
+  capplet_set_icon (w, "preferences-desktop-theme");
   gtk_widget_show_all (w);
 
   g_signal_connect_after (w, "response",
diff -up gnome-control-center-2.23.4/capplets/appearance/data/gnome-appearance-properties.desktop.in.in.standard-icon gnome-control-center-2.23.4/capplets/appearance/data/gnome-appearance-properties.desktop.in.in
--- gnome-control-center-2.23.4/capplets/appearance/data/gnome-appearance-properties.desktop.in.in.standard-icon	2008-06-20 14:03:23.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/appearance/data/gnome-appearance-properties.desktop.in.in	2008-06-20 14:03:38.000000000 -0400
@@ -2,7 +2,7 @@
 _Name=Appearance
 _Comment=Customize the look of the desktop
 Exec=gnome-appearance-properties %F
-Icon=gnome-settings-theme
+Icon=preferences-desktop-theme
 Terminal=false
 Type=Application
 StartupNotify=true
diff -up gnome-control-center-2.23.4/capplets/keybindings/gnome-keybinding-properties.c.standard-icon gnome-control-center-2.23.4/capplets/keybindings/gnome-keybinding-properties.c
--- gnome-control-center-2.23.4/capplets/keybindings/gnome-keybinding-properties.c.standard-icon	2008-06-20 13:53:42.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/keybindings/gnome-keybinding-properties.c	2008-06-20 13:54:16.000000000 -0400
@@ -1334,7 +1334,7 @@ setup_dialog (GladeXML *dialog)
   g_free (wm_name);
 
   widget = WID ("gnome-keybinding-dialog");
-  capplet_set_icon (widget, "gnome-settings-keybindings");
+  capplet_set_icon (widget, "preferences-desktop-keyboard-shortcuts");
   gtk_widget_show (widget);
 
   g_signal_connect (widget, "key_press_event", G_CALLBACK (maybe_block_accels), NULL);
diff -up gnome-control-center-2.23.4/capplets/keybindings/keybinding.desktop.in.in.standard-icon gnome-control-center-2.23.4/capplets/keybindings/keybinding.desktop.in.in
--- gnome-control-center-2.23.4/capplets/keybindings/keybinding.desktop.in.in.standard-icon	2008-06-20 13:52:32.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/keybindings/keybinding.desktop.in.in	2008-06-20 13:52:53.000000000 -0400
@@ -2,7 +2,7 @@
 _Name=Keyboard Shortcuts
 _Comment=Assign shortcut keys to commands
 Exec=gnome-keybinding-properties
-Icon=gnome-settings-keybindings
+Icon=preferences-desktop-keyboard-shortcuts
 Terminal=false
 Type=Application
 StartupNotify=true
diff -up gnome-control-center-2.23.4/capplets/keyboard/gnome-keyboard-properties.c.standard-icon gnome-control-center-2.23.4/capplets/keyboard/gnome-keyboard-properties.c
--- gnome-control-center-2.23.4/capplets/keyboard/gnome-keyboard-properties.c.standard-icon	2008-06-20 14:13:50.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/keyboard/gnome-keyboard-properties.c	2008-06-20 14:14:33.000000000 -0400
@@ -312,7 +312,7 @@ main (int argc, char **argv)
 		}
 
 		capplet_set_icon (WID ("keyboard_dialog"),
-				  "gnome-dev-keyboard");
+				  "preferences-desktop-keyboard");
 		gtk_widget_show (WID ("keyboard_dialog"));
 		gtk_main ();
 	}
diff -up gnome-control-center-2.23.4/capplets/keyboard/keyboard.desktop.in.in.standard-icon gnome-control-center-2.23.4/capplets/keyboard/keyboard.desktop.in.in
--- gnome-control-center-2.23.4/capplets/keyboard/keyboard.desktop.in.in.standard-icon	2008-06-20 14:13:27.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/keyboard/keyboard.desktop.in.in	2008-06-20 14:15:58.000000000 -0400
@@ -2,7 +2,7 @@
 _Name=Keyboard
 _Comment=Set your keyboard preferences
 Exec=gnome-keyboard-properties
-Icon=gnome-dev-keyboard
+Icon=preferences-desktop-keyboard
 Terminal=false
 Type=Application
 StartupNotify=true
diff -up gnome-control-center-2.23.4/capplets/windows/gnome-window-properties.c.standard-icon gnome-control-center-2.23.4/capplets/windows/gnome-window-properties.c
--- gnome-control-center-2.23.4/capplets/windows/gnome-window-properties.c.standard-icon	2008-06-20 14:18:59.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/windows/gnome-window-properties.c	2008-06-20 14:19:16.000000000 -0400
@@ -467,7 +467,7 @@ main (int argc, char **argv)
                 ++i;
         }
 
-        capplet_set_icon (dialog_win, "gnome-window-manager");
+        capplet_set_icon (dialog_win, "preferences-system-windows");
         gtk_widget_show (dialog_win);
 
         gtk_main ();
diff -up gnome-control-center-2.23.4/capplets/windows/window-properties.desktop.in.in.standard-icon gnome-control-center-2.23.4/capplets/windows/window-properties.desktop.in.in
--- gnome-control-center-2.23.4/capplets/windows/window-properties.desktop.in.in.standard-icon	2008-06-20 14:18:39.000000000 -0400
+++ gnome-control-center-2.23.4/capplets/windows/window-properties.desktop.in.in	2008-06-20 14:18:52.000000000 -0400
@@ -2,7 +2,7 @@
 _Name=Windows
 _Comment=Set your window properties
 Exec=gnome-window-properties
-Icon=gnome-window-manager
+Icon=preferences-system-windows
 Terminal=false
 Type=Application
 StartupNotify=true
